__tests__/apps.test.js

```diff
@@ -195,3 +197,63 @@ describe("4. PATCH /api/articles/:article_id", () => {
       });
   });
 });
+
+describe("5. GET /api/articles", () => {
+  test("status:200, returns all the articles if query is omitted", () => {
+    return request(app)
+      .get(`/api/articles`)
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toBeInstanceOf(Array);
+        expect(body.articles).toHaveLength(12);
+        let articles = body.articles;
+        articles.forEach((article) => {
+          expect(article).toEqual(
+            expect.objectContaining({
+              author: expect.any(String),
+              title: expect.any(String),
+              article_id: expect.any(Number),
+              topic: expect.any(String),
+              created_at: expect.any(String),
+              votes: expect.any(Number),
+            })
+          );
+        });
+      });
+  });
+  test("status:200, articles are sorted by date in descending order by default", () => {
+    return request(app)
+      .get("/api/articles?sort_by=created_at")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body;
+        expect(articles).toBeSortedBy("created_at", { descending: true });
+      });
+  });
+  test("status:200, articles are sorted by topic in descending order", () => {
+    return request(app)
+      .get("/api/articles?sort_by=topic")
```

Comment:
here in the path for this query, we want to specify which "topic" to sort by, for example "/api/articles?sort_by=mitch" or "/api/articles?sort_by=sport"

Reply:
> Hi Moroti, check my comments on the test file, it looks like we're testing for a sort_by instead of a filter, it should be ?topic= , this will mean changing the code around a bit to make sure it's doing the right thing

Okay,got confused as well. Was about to reply on that but working on it